src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,710 @@
+                    this.bondArray[tmpBondInteger].setProperty(AlkylStructureFragmenter.INTERNAL_ASF_FRAGMENTATION_PLACEMENT_KEY, false);
+                }
+            }
+        } catch (Exception anException) {
+            AlkylStructureFragmenter.this.logger.log(Level.WARNING,
+                    anException + " MoleculeID: " + anAtomContainer.getID(), anException);
+            throw new IllegalArgumentException("Unexpected error occurred during fragmentation of molecule: "
+                    + anAtomContainer.getID() + ", at cyclefinder : " + anException.toString());
+        }
+        //</editor-fold>
+        //
+    }
+
+    /**
+     * Private method to mark all atoms and bonds of any conjugated pi systems in the given atomcontainer.
+     *
+     * @param anAtomContainer IAtomContainer to mark atoms and bonds in
+     */
+    private void markConjugatedPiSystems(IAtomContainer anAtomContainer) {
+        //<editor-fold desc="ConjugatedPiSystemsDetector" defaultstate="collapsed">
+        try {
+            IAtomContainerSet tmpConjugatedAtomContainerSet;
+            tmpConjugatedAtomContainerSet = ConjugatedPiSystemsDetector.detect(anAtomContainer);
+            //molecule mapping
+            //iterate over every atomcontainer from ConjPiSystemsDetector output
+            for (IAtomContainer tmpConjAtomContainer: tmpConjugatedAtomContainerSet.atomContainers()) {
+                for (IAtom tmpConjAtom: tmpConjAtomContainer.atoms()) {
+                    int tmpAtomInteger = tmpConjAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_ATOM_INDEX_PROPERTY_KEY);
+                    this.atomArray[tmpAtomInteger].setProperty(AlkylStructureFragmenter.INTERNAL_ASF_FRAGMENTATION_PLACEMENT_KEY, false);
+                }
+                for (IBond tmpConjBond: tmpConjAtomContainer.bonds()) {
+                    int tmpBondInteger = tmpConjBond.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_BOND_INDEX_PROPERTY_KEY);
+                    this.bondArray[tmpBondInteger].setProperty(AlkylStructureFragmenter.INTERNAL_ASF_FRAGMENTATION_PLACEMENT_KEY, false);
+                }
+            }
+        } catch (Exception anException) {
+            AlkylStructureFragmenter.this.logger.log(Level.WARNING,
+                    anException + " MoleculeID: " + anAtomContainer.getID(), anException);
+            throw new IllegalArgumentException("Unexpected error occurred during fragmentation of molecule: "
+                    + anAtomContainer.getID() + " at conjugated pi systems detector: " + anException.toString());
+        }
+        //</editor-fold>
+    }
+
+    /**
+     * Private method to check given atomcontainer for disconnected structures.
+     *
+     * @param anAtomContainer IAtomContainer to check
+     * @return IAtomContainerSet containing partitioned structures as single IAtomContainer
+     */
+    private IAtomContainerSet checkConnectivity(IAtomContainer anAtomContainer) {
```

Comment:
method name and doc do not say correctly what this method does (separating unconnected fragments)